### DISCUSSION TITLE
similarity syntax with ffmpeg-python but have different output result

### DISCUSSION THREAD

**Discussion by sugizo** [livingbio/typed-ffmpeg#316] [Q&A] [answered]
**env**
google colab

**case** 
1. concat demuxer 

1.1. typed-ffmpeg
**steps**
```
pip install -U typed-ffmpeg
import ffmpeg
import IPython
(
    ffmpeg
    .input('http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4')
    .output(filename = 'output_t.mp4',
            t = 3, )
    .overwrite_output()
    .run()
)
in_file = ffmpeg.input('http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4')
ffmpeg.output(in_file.trim(duration = 3),
              in_file.atrim(duration = 3),
              filename = "trim_atrim_duration.mp4",).overwrite_output().run()
f = open('files.txt', "w")
f.write("""
file 'output_t.mp4'
file 'trim_atrim_duration.mp4'
""")
f.close()
(
    ffmpeg
    .input('files.txt', f = 'concat', safe = 0)
    .output(filename = 'concat_demuxer.mp4', c = 'copy')
    .overwrite_output()
    .run()
)
IPython.display.Video('concat_demuxer.mp4', embed = True)
```

no error occured for video concat_demuxer.mp4
but
**the result is not expected**
duration is 21 sec, 
and 
can be play only first 2 sec

**expected result**
duration is 6 sec (because output_t.mp4 duration is 3 sec
and
trim_atrim_duration.mp4 duration is 3 sec)
and can be play full duration 6 sec

1.2. ffmpeg-python
**steps**
```
pip install -U ffmpeg-python
import ffmpeg
import IPython
(
    ffmpeg
    .input('http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4')
    .output(filename = 'output_t.mp4',
            t = 3, )
    .overwrite_output()
    .run()
)
in_file = ffmpeg.input('http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4')
ffmpeg.output(in_file.trim(duration = 3),
              in_file.filter('atrim', duration = 3),
              filename = "trim_filter_atrim_duration.mp4"
).overwrite_output().run()
f = open('files.txt', "w")
f.write("""
file 'output_t.mp4'
file 'trim_filter_atrim_duration.mp4'
""")
f.close()
(
    ffmpeg
    .input('files.txt', f = 'concat', safe = 0)
    .output(filename = 'concat_demuxer.mp4', c = 'copy')
    .overwrite_output()
    .run()
)
IPython.display.Video('concat_demuxer.mp4', embed = True)
```

result output video is expected from ffmpeg-python which are 6 secs
and 
can be play full

any ideas what's going on in this matter ?

thanks and best regards

**Comment by sugizo** [comment]
problem spotted is on typed-ffmpeg part when execute
```
in_file = ffmpeg.input('http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4')
ffmpeg.output(in_file.trim(duration = 3),
              in_file.atrim(duration = 3),
              filename = "trim_atrim_duration.mp4",).overwrite_output().run()
IPython.display.Video('trim_atrim_duration.mp4', embed = True)
```

when executed and play the video is fine have a sound and duration is as expected 3 secs
but
when this file is concat the disaster will begin
as mention on the first posted

trying several tested using concat demuxer on typed-ffmpeg 
```
f = open('files.txt', "w")
f.write("""
file 'output_t.mp4'
file 'output_t.mp4'
""")
f.close()
```

the different from 1st post is the file for second video is different, 
when using output_t.mp4 the concat demuxer generated by typed-ffmpeg is 
expected result which 
duration concat_demuxer.mp4 is 6 secs 
and can be play full duration

so not sure which part is wrong on typed-ffmpeg when using trim() or atrim()
and combine it with concat demuxer

**Reply by lucemia** [reply]
Thank you for your report. I've created an issue to track this problem: [Issue #317](https://github.com/livingbio/typed-ffmpeg/issues/317).

It appears that FFmpeg behaves differently when an additional `-map` option is included:

FFmpeg command generated by ffmpeg-python:
`ffmpeg -i http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4 -t 3 output_t.mp4 -y`

FFmpeg command generated by typed-ffmpeg:
`ffmpeg -y -i http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4 -map 0 -t 3 output_t.mp4`

I will investigate further to determine the cause of this discrepancy.

**Comment by lucemia** [comment, marked as answer]
In the latest release, version 1.0.2, the issue has been resolved. The outputs of typed-ffmpeg and ffmpeg-python are now consistent.
